doc/examples_sphinx-gallery/personalized_pagerank.py

```diff
@@ -0,0 +1,81 @@
+"""
+.. _tutorials-personalized_pagerank:
+
+===============================
+Personalized PageRank on a grid
+===============================
+
+This example demonstrates how to calculate and visualize personalized PageRank on a grid. We use the :meth:`igraph.GraphBase.personalized_pagerank` method, and demonstrate the effects on a grid graph.
+"""
+
+import matplotlib.cm as cm
+import matplotlib.pyplot as plt
+import numpy as np
+
+import igraph as ig
+
+
+# %%
+# We define a function that plots the graph on a Matplotlib axis, along with
+# its personalized PageRank values. The function also generates a
+# color bar on the side to see how the values change.
+# We use `Matplotlib's Normalize class <https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.Normalize.html>`_
+# to set the colors and ensure that our color bar range is correct.
+def plot_pagerank(graph: ig.Graph, p_pagerank: list[float]):
+    """Plots personalized PageRank values on a grid graph with a colorbar.
+
+    Parameters
+    ----------
+    graph : ig.Graph
+        graph to plot
+    p_pagerank : list[float]
+        calculated personalized PageRank values
+    """
+    # Create the axis for matplotlib
+    _, ax = plt.subplots(figsize=(8, 8))
+
+    # Create a matplotlib colormap
+    # coolwarm goes from blue (lowest value) to red (highest value)
+    cmap = cm.coolwarm
+
+    # Normalize the PageRank values for colormap
+    normalized_pagerank = ig.rescale(p_pagerank)
+
+    graph.vs["color"] = [cmap(pr) for pr in normalized_pagerank]
+    graph.vs["size"] = ig.rescale(p_pagerank, (20, 40))
+    graph.es["color"] = "gray"
+    graph.es["width"] = 1.5
+
+    # Plot the graph
+    ig.plot(graph, target=ax, layout=graph.layout_grid())
+
+    # Add a colorbar
+    sm = cm.ScalarMappable(norm=plt.Normalize(min(p_pagerank), max(p_pagerank)), cmap=cmap)
+    plt.colorbar(sm, ax=ax, label="Personalized PageRank")
+
+    plt.title("Graph with Personalized PageRank")
+    plt.show()
```

Comment:
Let's make the plot not be stretched vertically or horizontally. Something like `plt.axis("equal")` can achieve this.